### PR TITLE
Encode binary data as hex literal in backup dump

### PR DESCRIPTION
**Pull request by ausi**
This should improve compatibility with SQL tools like SequelPro or SequelAce.

### CODE REVIEW

**Comment by aschempp** [comment]
I understand why to use `bin2hex` but I don't understand why we should stop to use DBAL (to create view and detect the type)?

**Comment by ausi** [comment]
> I don't understand why we should stop to use DBAL (to create view…

`CREATE VIEW` does not drop the view before (re-)creating it. That’s why I changed it to `CREATE OR REPLACE VIEW`. DBAL doesn’t support that.

> …and detect the type)?

The type of the column is not relevant actually. Every non-UTF-8 column has to be encoded if it contains non-ASCII data in order to be compatible with several SQL tools. And every UTF-8 column needs to be encoded if it contains non-UTF8 data.

File: core-bundle/src/Doctrine/Backup/Dumper.php [hunk 1]
```diff
@@ -61,7 +60,7 @@ private function dumpViews(AbstractSchemaManager $schemaManager, AbstractPlatfor
     {
         foreach ($schemaManager->listViews() as $view) {
             yield sprintf('-- BEGIN VIEW %s', $view->getName());
-            yield $platform->getCreateViewSQL($view->getQuotedName($platform), $view->getSql()).';';
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/Dumper.php:64]: We can just make 2 queries out of it, as it's non persistent data anyway? Like so?

```php
            yield $platform->getDropViewSQL($view->getQuotedName($platform)).';';
            yield $platform->getCreateViewSQL($view->getQuotedName($platform), $view->getSql()).';';
```

File: core-bundle/src/Doctrine/Backup/Dumper.php [hunk 2]
```diff
@@ -117,8 +112,16 @@ private function formatValueForDump($value, Column $column, Connection $connecti
             return 'NULL';
         }
 
-        if ($column->getType() instanceof BinaryType) {
-            return sprintf('UNHEX(%s)', $connection->quote($value));
+        // non-ASCII values
+        if (
+            \is_string($value)
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/Dumper.php:117]: Can we have a test for that in `DumperTest` maybe? ☺️

File: core-bundle/src/Doctrine/Backup/Dumper.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Doctrine/Backup/Dumper.php:64]: This will fail if the view does not exist.

Comment (Toflar) [reply core-bundle/src/Doctrine/Backup/Dumper.php:64]: Right :) 👍 then

File: core-bundle/src/Doctrine/Backup/Dumper.php (hunk 2 above)
Comment (ausi) [reply core-bundle/src/Doctrine/Backup/Dumper.php:117]: Done in 0733330e9b7bb40bacfdc1ef19815fe785673de8

**Review by Toflar — APPROVED** [review]
Thanks for the tests <3

**Review by m-vo — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**